autolink/postlink/activityLinker.js

```diff
@@ -1,23 +1,52 @@
 // @ts-check
-var path = require('./path');
+var path = require("./path");
 var fs = require("fs");
-var { warnn, logn, infon, debugn } = require("./log");
+var { errorn, warnn, logn, infon, debugn } = require("./log");
 
 class ActivityLinker {
   constructor() {
     this.activityPath = path.mainActivityJava;
+    this.extendNavigationActivity = false;
+    this.removeGetMainComponentName = false;
   }
 
   link() {
+    if (!this.activityPath) {
+      errorn(
+        "   MainActivity not found! Does the file exist in the correct folder?\n   Please check the manual installation docs:\n   https://wix.github.io/react-native-navigation/docs/installing#2-update-mainactivityjava"
+      );
+      return;
+    }
+
     logn("Linking MainActivity...");
-    if (this.activityPath) {
-      var activityContent = fs.readFileSync(this.activityPath, "utf8");
+
+    var activityContent = fs.readFileSync(this.activityPath, "utf8");
+
+    try {
       activityContent = this._extendNavigationActivity(activityContent);
+      this.extendNavigationActivity = true;
+    } catch (e) {
+      errorn("   " + e);
```

Comment:
Here we are concatenating a string with an `Error` type variable, so `e` will by converted to string by doing `e.toString()`, resulting in something like `Error: $MESSAGE'.

Maybe we should be explicit in what we want to output to `errorn`, such as :
```js
      errorn("   " + e.message);
```